src/frontend/js/search-chats.js

```diff
@@ -0,0 +1,240 @@
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      max-width: calc(100% - 80px);
+    }
+
+ 
+    .dropdown .nofound {
+        color: gray;
+        font-size: 10pt;
+        align-self: center;
+        font-weight: lighter;
+    }
+
+
+    .material-icons {
+      font-family: "Material Icons";
+      font-weight: normal;
+      font-style: normal;
+      font-size: 24px;
+      line-height: 1;
+      letter-spacing: normal;
+      text-transform: none;
+      display: inline-block;
+      white-space: nowrap;
+      word-wrap: normal;
+      direction: ltr;
+      -webkit-font-smoothing: antialiased;
+      cursor: pointer;
+    }
+
+    .dropdown > div:hover {
+      background: lightgray;
+    }
+
+    #searchChats .containerInput {
+      position: relative;
+    }
+
+    .containerInput input:focus ~ span {
+      opacity: 0;
+      visibility: hidden;
+    }
+
+    .containerInput span {
+      position: absolute;
+      transform: translate(-50%, -50%);
+      top: 50%;
+      right: 0px;
+      z-index: 5;
+      color: #6f6f6f;
+      transition: 0.5s;
+    }
+
+    .dropdown .avatar {
+      background: gray;
+    }
+
+    .avatar {
+      width: 50px;
+      height: 50px;
+      border-radius: 50%;
+      
+    }
+
+    .conversazioni:has(#searchChats input:focus) .elencoFarmacie {
+      opacity: 0;
+    }
+
+    `
+
+    constructor() {
+        super()
+        this.elencoFarmacie = []
+        this.query = ''
+    }
+
+    render() {
+        return html`
+        <div id="searchChats">
+            <div class="containerInput">
+
+                <input type="text" placeholder="cerca farmacie" @input=${this.filtra} />
+                <span class="material-icons"> search </span>
+
+                <div class="dropdown">
+                        ${
+                            this.mostraSuggerimenti()
+                        }
+                </div>
+
+            </div>
+        </div>
+        `
+    }
+
+    filtra(event) {
+        const text = event.target.value;
+        this.query = text.toLowerCase()
+
+        const farmacie = this.myfetch()
+        let tmp = [] 
+        
+        farmacie.forEach( farmacia => {
+            if(farmacia.name.toLowerCase().indexOf(this.query) > -1)
+                tmp.push(farmacia)
+        })
+
+        this.elencoFarmacie = tmp
+
+    }
+
+    myfetch() {
+     
+        /*
+        Dato che non c'è neancora l'API per prendere i dati dal server simulo una funzione che
+        ritorno con una chiamata HTTP un json parserizzato
+        */ 
+
+        const json =  
+        [
+            { name: 'farmacia1', avatar: './fakeServer/avatar1.png'},
+            { name: 'farmacia2', avatar: './fakeServer/avatar1.png'},
+            { name: 'farmacia3', avatar: './fakeServer/avatar1.png'},
+            { name: 'farmacia del Dr. Fantozzi Ugo', avatar: './fakeServer/avatar1.png'},
+        ]        
+
+        return json
+    }
+
+
+    mostraSuggerimenti() {
```

Comment:
usiamo l'inglese per i nomi di classi, funzioni o variabili

Reply:
cambiato in showTips()